generate_tag/action.yaml

```diff
@@ -0,0 +1,35 @@
+# This action expects the git event data to be stored as an environment variable by the calling workflow
+
+name: phil_generate_tag
+inputs:
+  sha:
+    description: "Shortened commit SHA"
+    required: true
+outputs:
+  tag:
+    description: "Generated tag based on Git event"
+    value: ${{ steps.generate-image-tag.outputs.tag }}
+runs:
+  using: "composite"
+  steps:
+    - id: generate-image-tag
```

Comment:
Let's try simplifying this so it's more readable. I think multiple steps would help, since we can label the steps and give more context of what each piece is achieving.

```yaml
runs:
  using: "composite"
  steps:
    - name: If the ref is a tag, remove the leading v
       if:  startsWith(${GITHUB_REF_NAME} , "v")
       run: echo "::set-output name=TAG::$( echo $GITHUB_REF_NAME | sed -e 's/v//1' )

    - name: If the ref is feature branch, lowercase it and replace all special chars with hyphens
      if: startsWith(${GITHUB_REF_NAME}, "feature/")
      run: echo "::set-output name=TAG::$( echo $GITHUB_REF_NAME | sed -e 's/feature\///g' -e 's/_/-g' | awk '{print tolower($0)}')

    - name: If the ref is master/relase, set the tag to the first 8 of the commit sha
       if: ${GITHUB_REF_NAME} == "master" || ${GITHUB_REF_NAME} == "release"
       run: echo "::set-ouput name=TAG::${GITHUB_SHA:8}
```